openfl/utilities/utils.py

```diff
@@ -202,11 +207,32 @@ def validate_file_hash(file_path, expected_hash, chunk_size=8192):
 
 def tqdm_report_hook():
     """Visualize downloading."""
+
     def report_hook(pbar, count, block_size, total_size):
         """Update progressbar."""
         if pbar.total is None and total_size:
             pbar.total = total_size
         progress_bytes = count * block_size
         pbar.update(progress_bytes - pbar.n)
+
     pbar = tqdm(total=None)
     return partial(report_hook, pbar)
+
+
+def merge_configs(
+        overwrite_dict: Optional[dict] = None,
+        data_preparation: Optional[List[Tuple[str, Callable]]] = None,
+        **kwargs,
+) -> Dynaconf:
+    """Create Dynaconf settings, merge its with `overwrite_dict` and validate result."""
+    settings = Dynaconf(**kwargs, YAML_LOADER='safe_load')
+    if overwrite_dict:
+        for key, value in overwrite_dict.items():
+            if value is not None or settings.get(key) is None:
+                settings.set(key, value, merge=True)
+    if data_preparation:
```

Comment:
This is interesting. Because the purpose is transforming the provided value in some way to match the expected format, I think this could be generalized beyond `data_preparation` (an example would be a malformed URL provided by the user in the case that `merge_configs` was extended to other functions). Maybe change the name to `value_transform`?